src/main.rs

```diff
@@ -93,6 +93,8 @@ fn main() -> Result<()> {
     std::fs::set_permissions(path, std::fs::Permissions::from_mode(0o777))?;
     spawn_acceptor(&mut wg, &logger, listener, tx, handoff_timeout);
 
+    let _ = sd_notify::notify(true, &[NotifyState::Ready]);
```

Comment:
Was curious about this but a) that's the [example in the Rust crate](https://docs.rs/sd-notify/latest/sd_notify/) and b) [the C docs](https://www.freedesktop.org/software/systemd/man/sd_notify.html) specifically say, "...it is generally recommended to ignore the return value of this call."

Reply:
See https://docs.rs/sd-notify/latest/src/sd_notify/lib.rs.html#105

You don't want to fail if informing about readyness failed, e.g. if the service is /not/ run via systemd and/or the communication with the unix socket not possible.

See also https://gist.github.com/grawity/6e5980981dccf66f554bbebb8cd169fc.